totalRP3_Extended/aura/aura.lua

```diff
@@ -0,0 +1,1009 @@
+				category = "|cff" ..aura.color.h .. aura.class.BA.CA .. "|r";
+			else
+				category = aura.class.BA.CA;
+			end
+		end
+		
+		if aura.class.BA.DE then
+			description = TRP3_API.script.parseArgs(aura.class.BA.DE, { object = aura.persistent });
+		end
+		
+		if aura.class.BA.FL then
+			flavor = aura.class.BA.FL;
+		end
+		
+		if aura.persistent.expiry < math.huge then
+			expiry = loc.AU_EXPIRY:format(self.timeFormatterNormal:Format(aura.persistent.expiry - self:Now()));
+		end
+		
+		if aura.class.BA.CC then
+			cancelText = loc.AU_CANCEL_TEXT;
+		end
+		
+		return title, category, description, flavor, expiry, cancelText;
+	end,
+	
+	GetAurasForInspection = function(self)
+		local auras = {};
+		for i = 1,#self.activeAuras do
+			local aura = self.activeAuras[i];
+			if not aura.persistent.invalid and aura.class.BA.WE then
+				tinsert(auras, {
+					class = {
+						BA = {
+							NA = aura.class.BA.NA,
+							CA = aura.class.BA.CA,
+							DE = TRP3_API.script.parseArgs(aura.class.BA.DE, { object = aura.persistent }),
+							OV = TRP3_API.script.parseArgs(aura.class.BA.OV, { object = aura.persistent }),
+							IC = aura.class.BA.IC,
+							HE = aura.class.BA.HE,
+							FL = aura.class.BA.FL,
+							CO = aura.class.BA.CO,
+						}						
+					},
+					persistent = {
+						expiry = aura.persistent.expiry or math.huge
+					}
+				});
+			end
+		end
+		return auras;
+	end,
+	
+	UpdateInspectionFrame = function(self, auras)
+		self.inspectionAuraFramePool:ReleaseAll();
+		
+		if not auras then return end
+		
+		local buffNum, debuffNum = 0, 0;
+		local x, y;
+		
+		for i = 1,#auras do
+			local frame = self.inspectionAuraFramePool:Acquire();
+			if auras[i].class.BA.HE then
+				x =  INSPECT_AURA_OFFSET + math.floor(buffNum/INSPECT_AURAS_PER_COL)*INSPECT_AURA_COL_WIDTH;
+				y = -INSPECT_AURA_OFFSET - (buffNum % INSPECT_AURAS_PER_COL) * INSPECT_AURA_HEIGHT;
+				frame:SetPoint("TOPLEFT", TRP3_InspectionFrame.Main.Model, "TOPLEFT", x, y);
+				buffNum = buffNum + 1;
+			else
+				x = -INSPECT_AURA_OFFSET - math.floor(debuffNum/INSPECT_AURAS_PER_COL)*INSPECT_AURA_COL_WIDTH;
+				y = -INSPECT_AURA_OFFSET - (debuffNum % INSPECT_AURAS_PER_COL) * INSPECT_AURA_HEIGHT;
+				frame:SetPoint("TOPRIGHT", TRP3_InspectionFrame.Main.Model, "TOPRIGHT", x, y);
+				debuffNum = debuffNum + 1;
+			end
+			frame.icon:SetTexture("Interface\\ICONS\\" .. (auras[i].class.BA.IC or "TEMP"));
+			frame.overlay:SetText(auras[i].class.BA.OV);
+			frame.duration:SetText("");
+			auras[i].persistent.expiry = auras[i].persistent.expiry or math.huge;
+			auras[i].color = self:GetAuraColorFromClass(auras[i].class);
+			if auras[i].color then
+				frame.border:SetVertexColor(auras[i].color.r, auras[i].color.g, auras[i].color.b);
+				frame.border:Show();
+			else
+				frame.border:Hide();
+			end
+			frame.aura = auras[i];
+			auras[i].frame = frame;
+			frame:Show();
+		end
+		
+	end,
+}
+
+TRP3_API.extended.auras.apply = function(auraId, extend)
```

Comment:
I'm not a fan of the Extend option on its own.
I think either all duration operations must go through the Aura duration effect and the checkbox should be removed, or the checkbox should be replaced by a dropdown adding Refresh (so Do nothing/Extend/Refresh), as it would be a common buff behaviour.